overops-functions/src/main/java/com/takipi/udf/microsoftteams/card/MicrosoftCard.java

```diff
@@ -0,0 +1,103 @@
+package com.takipi.udf.microsoftteams.card;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MicrosoftCard {
+    @SerializedName("@type")
+    public String type;
+    @SerializedName("@context")
+    public String context;
+    public String themeColor;
+    public String summary;
+    public String text;
+    public List<MicrosoftSection> sections = new ArrayList<>();
+    public List<MicrosoftPotentialAction> potentialAction = new ArrayList<>();
+
+    public MicrosoftCard(String type, String context, String themeColor, String summary, String text, List<MicrosoftSection> sections, List<MicrosoftPotentialAction> potentialAction) {
+        this.type = type;
+        this.context = context;
+        this.themeColor = themeColor;
+        this.summary = summary;
+        this.text = text;
+        this.sections = sections;
+        this.potentialAction = potentialAction;
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        public String type = "MessageCard";
+        public String context = "http://schema.org/extensions";
+        public String themeColor = "0076D7";
+        public String summary = "Summary";
+        public String text = "";
+        public List<MicrosoftSection> sections = new ArrayList<>();
+        public List<MicrosoftPotentialAction> potentialAction = new ArrayList<>();
+
+        public Builder setType(String type) {
+            this.type = type;
+            return this;
+        }
+
+        public Builder setContext(String context) {
+            this.context = context;
+            return this;
+        }
+
+        public Builder setThemeColor(String themeColor) {
+            this.themeColor = themeColor;
+            return this;
+        }
+
+        public Builder setSummary(String summary) {
+            this.summary = summary;
+            return this;
+        }
+
+        public Builder setText(String text) {
+            this.text = text;
+            return this;
+        }
+
+        public Builder setSections(List<MicrosoftSection> sections) {
+            this.sections = sections;
+            return this;
+        }
+
+        public Builder addSections(MicrosoftSection... sections) {
+            if (this.sections == null) this.sections = new ArrayList<>();
```

Comment:
Please have {} surrounding if code blocks, even if it's just a single line.
In this case:

```
if (this.sections == null) {
    this.sections = new ArrayList<>();
}
```

Same for other methods in this class

Reply:
done